snapcraft/parts/lifecycle.py

```diff
@@ -255,6 +255,7 @@ def _run_command(
         not managed_mode
         and not parsed_args.destructive_mode
         and not os.getenv("SNAPCRAFT_BUILD_ENVIRONMENT") == "host"
+        or parsed_args.use_lxd
     ):
```

Comment:
This `if` statement is not clean.

It could be rewritten as
```python
if not (
    (
        managed_mode
        or parsed_args.destructive_mode
        or os.getenv("SNAPCRAFT_BUILD_ENVIRONMENT") == "host"
    )
    and not parsed_args.use_lxd
):
```

or I could break up this whole function up (in a separate refactoring PR) to make it more readable.  Or I can leave it as-is.

Reply:
The version with explicit precedence is safer for future refactorings and easier to read, but we can also set a variable based on `use_lxd` and `SNAPCRAFT_BUILD_ENVIRONMENT`, and then use this variable in the conditional.